include/glow/Base/Type.h

```diff
@@ -78,6 +78,41 @@ struct ShapeNHWC {
   }
 };
 
+struct ShapeNCHW {
+  size_t n; // Number of samples
+  size_t c; // Number of Channels
+  size_t h; // Height
+  size_t w; // Width
+
+  explicit ShapeNCHW(llvm::ArrayRef<size_t> shape) {
+    assert(shape.size() == 4 && "Invalid shape");
+    n = shape[0];
+    c = shape[1];
+    h = shape[2];
+    w = shape[3];
+  }
+
+  static ShapeNCHW fromXYZ(llvm::ArrayRef<size_t> shape) {
```

Comment:
does that imply construction happens based on NHW where C = 1. If so, maybe name that appropriately?